File: fix_ts.js
const fs = require('fs');
const path = require('path');

const toolsDir = path.join('c:', 'projects', 'alvion-next', 'src', 'app', 'financial-tools');

function processDirectory(dir) {
    const files = fs.readdirSync(dir);
    
    for (const file of files) {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        
        if (stat.isDirectory()) {
            processDirectory(filePath);
        } else if (file.endsWith('.tsx')) {
            processFile(filePath);
        }
    }
}

function processFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf-8');
    
    // Regex looks for setX(e.target.value)
    // We skip setCalcType explicitly
    const regex = /(set[A-Za-z0-9_]+)\(e\.target\.value\)/g;
    
    const newContent = content.replace(regex, (match, setter) => {
        if (setter === 'setCalcType') {
            return match;
        }
        return `${setter}(Number(e.target.value))`;
    });
    
    if (newContent !== content) {
        fs.writeFileSync(filePath, newContent, 'utf-8');
        console.log(`Fixed ${filePath}`);
    }
}

processDirectory(toolsDir);
console.log("Done fixing TypeScript errors!");

File: fix_ts.py
import os
import re

tools_dir = r"c:\projects\alvion-next\src\app\financial-tools"

# we only replace e.target.value with Number(e.target.value) for number states.
# since setCalcType is string, we skip it.
# we use a regex to replace: onChange={(e) => setXYZ(e.target.value)} -> onChange={(e) => setXYZ(Number(e.target.value))}

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find all setter names that are called with e.target.value
    # e.g., setYears(e.target.value)
    
    # We will exclude setCalcType explicitly
    def replacer(match):
        setter = match.group(1)
        if setter == "setCalcType":
            return match.group(0) # don't touch
        return f"{setter}(Number(e.target.value))"

    # Regex looks for setX(e.target.value)
    new_content = re.sub(r'(set[A-Za-z0-9_]+)\(e\.target\.value\)', replacer, content)

    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Fixed {filepath}")

for root, _, files in os.walk(tools_dir):
    for file in files:
        if file.endswith(".tsx"):
            process_file(os.path.join(root, file))

print("Done")
